spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/redis/DataRedisTestIntegrationTests.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.test.autoconfigure.data.redis;
+
+import java.nio.charset.Charset;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import redis.embedded.RedisServer;
+
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.core.RedisOperations;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+/**
+ * Integration test for {@link DataRedisTest}.
+ *
+ * @author Jayaram Pradhan
+ */
+@RunWith(SpringRunner.class)
+@DataRedisTest
+public class DataRedisTestIntegrationTests {
+
+	@Autowired
+	private RedisOperations<Object, Object> operations;
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Autowired
+	private ExampleRepository exampleRepository;
+
+	@Autowired
+	private ApplicationContext applicationContext;
+
+	private static final Charset CHARSET = Charset.forName("UTF-8");
+
+	private static RedisServer redisServer;
+
+	@BeforeClass
+	public static void setUp() throws Exception {
+		redisServer = new RedisServer(6379);
+		redisServer.start();
```

Comment:
@snicoll Thank you. yes I feel the same. Looks like there is a [PR](https://github.com/spring-projects/spring-boot/pull/6497) already open. potentially this could wait till it has been merged.